src/components/dragZoomLayer.ts

```diff
@@ -0,0 +1,152 @@
+      });
+
+      this._doubleClickInteraction.onDoubleClick(() => this.unzoom());
+    }
+
+    /* Set the time (in ms) over which the zoom will interpolate.
+     * 0 implies no interpolation. (ie zoom is instant)
+     */
+    public animationTime(): number;
+    public animationTime(animationTime: number): DragZoomLayer;
+    public animationTime(animationTime?: number): any {
+      if (animationTime == null) {
+        return this._animationTime;
+      }
+      if (animationTime < 0) {
+        throw new Error("animationTime cannot be negative");
+      }
+      this._animationTime = animationTime;
+      return this;
+    }
+
+    /* Set the easing function, which determines how the zoom interpolates over time. */
+    public ease(fn: (t: number) => number): DragZoomLayer {
+      if (typeof(fn) !== "function") {
+        throw new Error("ease function must be a function");
+      }
+      if (fn(0) !== 0 || fn(1) !== 1) {
+        Utils.Window.warn("Easing function does not maintain invariant f(0)==0 && f(1)==1. Bad behavior may result.");
+      }
+      this.easeFn = fn;
+      return this;
+    }
+
+    // Zoom into extent of the selection box bounds
+    private zoom() {
+      let x0: number = this.xExtent()[0].valueOf();
+      let x1: number = this.xExtent()[1].valueOf();
+      let y0: number = this.yExtent()[1].valueOf();
+      let y1: number = this.yExtent()[0].valueOf();
+
+      if (x0 === x1 || y0 === y1) {
+        return;
+      }
+
+      if (!this.isZoomed) {
+        this.isZoomed = true;
+        this.xDomainToRestore = this.xScale().domain();
+        this.yDomainToRestore = this.yScale().domain();
+      }
+      this.interpolateZoom(x0, x1, y0, y1);
+    }
+
+    // Restore the scales to their state before any zoom
+    private unzoom() {
+      if (!this.isZoomed) {
+        return;
+      }
+      this.isZoomed = false;
+      this.interpolateZoom(this.xDomainToRestore[0], this.xDomainToRestore[1],
+                           this.yDomainToRestore[0], this.yDomainToRestore[1]);
+    }
+
+    // If we are zooming, disable interactions, to avoid contention
+    private isZooming(isZooming: boolean) {
+      this._dragInteraction.enabled(!isZooming);
+      this._doubleClickInteraction.enabled(!isZooming);
+    }
+
+    private interpolateZoom(x0f: Numeric, x1f: Numeric, y0f: Numeric, y1f: Numeric) {
+      let x0s: number = this.xScale().domain()[0].valueOf();
+      let x1s: number = this.xScale().domain()[1].valueOf();
+      let y0s: number = this.yScale().domain()[0].valueOf();
+      let y1s: number = this.yScale().domain()[1].valueOf();
+
+      // Copy a ref to the ease fn, so that changing ease wont affect zooms in progress
+      let ease = this.easeFn;
+      let interpolator = (a: Numeric, b: Numeric, p: number): number => {
+        return d3.interpolateNumber(a.valueOf(), b.valueOf())(ease(p));
+      };
+      this.isZooming(true);
+      let start = Date.now();
+      let draw = () => {
```

Comment:
Wonder if we should be implementing our own animation logic here instead of using a Plottable.Animator